scheme/ocidir/ocidir.go

```diff
@@ -132,12 +132,14 @@ func (o *OCIDir) Throttle(r ref.Ref, put bool) []*throttle.Throttle {
 	if !put || o.throttleDef <= 0 {
 		return tList
 	}
-	o.mu.Lock()
-	defer o.mu.Unlock()
+
 	return []*throttle.Throttle{o.throttleGet(r)}
 }
 
 func (o *OCIDir) throttleGet(r ref.Ref) *throttle.Throttle {
```

Comment:
this is also used directly, need to lock

Reply:
I was keeping most of the locks in the exported methods, or passing a `locked bool` to the private methods if it needed to be handled at a lower level, to avoid recursive locking issues. For this one, I'd rather add a short lock in `BlobPut`, something like:

```
func (o *OCIDir) BlobPut(ctx context.Context, r ref.Ref, d types.Descriptor, rdr io.Reader) (types.Descriptor, error) {
	o.mu.Lock()
	t := o.throttleGet(r)
	o.mu.Unlock()
	err := t.Acquire(ctx)
	if err != nil {
		return d, err
	}
	defer t.Release(ctx)
```

Reply:
Actually I'm second guessing myself, lets do both the variable in BlobPut so the method is only called once, but pass a `locked bool` to the method so it's clear this method needs a lock. Then wrap the mutex calls in an `if !locked` like is done in other private methods.

Reply:
I see, ok. will fix.